src/pages/audio-rewards-page/store/sagas.ts

```diff
@@ -271,6 +273,16 @@ export function* watchFetchUserChallenges() {
         ) {
           newDisbursement = true
         }
+        if (
+          (challengeOverrides?.current_step_count ?? 0) > 0 &&
+          challenge.current_step_count !== 0
```

Comment:
Might make more sense to `put(updateOptimisticListenStreak)` on success and handle the logic there. I don't _think_ this should be general for every challenge type?